ksmt-core/src/main/kotlin/org/ksmt/expr/rewrite/simplify/KBvExprSimplifier.kt

```diff
@@ -128,19 +117,19 @@ interface KBvExprSimplifier : KExprSimplifierBase {
 
     override fun <T : KBvSort> transform(expr: KBvUnsignedLessOrEqualExpr<T>): KExpr<KBoolSort> =
         simplifyApp(expr) { (lhs, rhs) ->
-            bvLessOrEqual(lhs, rhs, signed = false)
+            simplifyBvUnsignedLessOrEqualExpr(lhs, rhs)
         }
 
     override fun <T : KBvSort> transform(expr: KBvSignedLessOrEqualExpr<T>): KExpr<KBoolSort> =
         simplifyApp(expr) { (lhs, rhs) ->
-            bvLessOrEqual(lhs, rhs, signed = true)
+            simplifyBvSignedLessOrEqualExpr(lhs, rhs)
         }
 
     // (uge a b) ==> (ule b a)
     override fun <T : KBvSort> transform(expr: KBvUnsignedGreaterOrEqualExpr<T>): KExpr<KBoolSort> =
         simplifyApp(
             expr = expr,
-            preprocess = { mkBvUnsignedLessOrEqualExpr(expr.arg1, expr.arg0) }
+            preprocess = { KBvUnsignedLessOrEqualExpr(this, expr.arg1, expr.arg0) }
```

Comment:
Why we used to have a function dependent on context, and now we have a raw constructor?

Reply:
We can use unmanaged object here and avoid interning since this object is guaranteed to be rewritten.